astroquery/eso/core.py

```diff
@@ -36,17 +42,41 @@ def _check_response(content):
         return True
 
 
-class EsoClass(QueryWithLogin):
+class CalSelectorError(Exception):
+    pass
+
+
+class AuthInfo:
+    def __init__(self, username: str, password: str, token: str):
+        self.username = username
+        self.password = password
+        self.token = token
+        self.expiration_time = self._get_exp_time_from_token()
+
+    def _get_exp_time_from_token(self) -> int:
+        # "manual" decoding since jwt is not installed
+        decoded_token = base64.b64decode(self.token.split(".")[1] + "==")
+        return int(json.loads(decoded_token)['exp'])
+
+    def expired(self) -> bool:
+        # we anticipate the expiration time by 10 minutes to avoid issues
+        return time.time() > self.expiration_time - 600
 
+
+class EsoClass(QueryWithLogin):
     ROW_LIMIT = conf.row_limit
     USERNAME = conf.username
     QUERY_INSTRUMENT_URL = conf.query_instrument_url
+    CALSELECTOR_URL = "https://archive.eso.org/calselector/v1/associations"
+    DOWNLOAD_URL = "https://dataportal.eso.org/dataPortal/file/"
+    AUTH_URL = "https://www.eso.org/sso/oidc/token"
+    GUNZIP = "gunzip"
```

Comment:
Actually I didn't foresee to change these values, that's why I've defined them as "constants". Regarding `gunzip`, we might change the handling of .Z files again in the future since we have now the possibility of changing compression on-the-fly upon download so we could return only .gz files to astroquery users.